AlfiyaSiddique/TastyTrails
# FEATURE: Recipe For The Day Suggestion section

Fixes #328

## What does this PR do?

This PR implements a "Recipe of the Day" feature for TastyTrails. It includes:

1. Backend changes:
   - New fields in the Recipe model: `featured` and `lastFeatured`
   - New controller function `getRecipeOfTheDay`
   - New route `/recipe-of-the-day`

2. Frontend changes:
   - New `RecipeOfTheDay` component
   - Integration of the new component in the Landing page

The feature selects a new recipe each day, avoiding repetition within 7 days when possible. It provides users with daily culinary inspiration directly on the landing page.

Fixes #328 (Implement Recipe of the Day feature)

## How should this be tested?

1. Backend Testing:
   - Start the backend server
   - Use a tool like Postman to send a GET request to `/api/recipe-of-the-day`
   - Verify that a recipe is returned with `featured: true`
   - Run this test on multiple days to ensure a different recipe is featured

2. Frontend Testing:
   - Start the frontend application
   - Navigate to the landing page
   - Scroll down to find the "Recipe of the Day" section
   - Verify that a recipe is displayed with its image, name, and description
   - Click the "View Recipe" button and confirm it navigates to the full recipe page
   - Check the responsiveness on different screen sizes

3. Long-term Testing:
   - Monitor for 7+ days to ensure recipes are not repeated within a week unless necessary

## Checklist

### Required
